modules/tensor_mechanics/src/materials/ComputeStrainBase.C

```diff
@@ -27,6 +27,8 @@ validParams<ComputeStrainBase>()
       "volumetric_locking_correction", false, "Flag to correct volumetric locking");
   params.addParam<std::vector<MaterialPropertyName>>(
       "eigenstrain_names", "List of eigenstrains to be applied in this strain calculation");
+  params.addParam<MaterialPropertyName>("global_strain",
+                                        "Material Propertyname for calculating global strain.");
```

Comment:
Let's make this

`Optional material property holding a global deformation tensor applied to the mesh as a whole`